### PR TITLE
3777.gbs immutable download spec improvements

### PR DESCRIPTION
**Pull request by exarkun**
https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3777

I opted to simplify the interface.  The points on the ticket about how shoving all of this complexity into a single HTTP request don't really make sense are pretty compelling.  I read enough of the immutable downloader code to convince myself that it only *actually* wants to read one region at a time anyway (that is, there is no client logic that would change from 1 request to N requests as a result of this interface change).  Maybe some *other* code could have a good reason for reading from multiple regions but we can cross that bridge when we get to it.

There's a note on the ticket about how the same may apply to the mutable download interface.  It ... may.  I haven't spent enough time with the implementation of mutables to be sure one way or the other, though.  Does the client really need to read from two different shares at once?  Does it need to read from two different regions of those shares?  I'm not sure...  It is maybe noteworthy that Tahoe makes a single `testv_and_readv_and_writev` atomic.  Does that matter?  Again, I'm not sure...

### CODE REVIEW

File: docs/proposed/http-storage-node-protocol.rst [hunk 1]
```diff
@@ -509,28 +509,21 @@ For example::
...
+``GET /v1/immutable/:storage_index/:share_number?offset=:offset&length=:length
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 
-Read data from the indicated immutable shares.
-If ``share`` query parameters are given, selecte only those shares for reading.
-Otherwise, select all shares present.
-If ``size`` and ``offset`` query parameters are given,
-only the portions thus identified of the selected shares are returned.
-Otherwise, all data is from the selected shares is returned.
+Read a contiguous sequence of bytes from one share in one bucket.
```
Comment (itamarst) [inline docs/proposed/http-storage-node-protocol.rst:515]: At this point maybe worth reconsidering `Range` requests, and say "multi-range requests need not be supported"?

**Review by itamarst — CHANGES_REQUESTED** [review]
Thank you! See the comment.

File: docs/proposed/http-storage-node-protocol.rst (hunk 1 above)
Comment (exarkun) [reply docs/proposed/http-storage-node-protocol.rst:515]: Hm yea I guess that might work.  I kinda thought HTTP required multipart response type for *all* range requests... but it looks like that's wrong and a single range can just receive raw resource bytes as the response (per https://datatracker.ietf.org/doc/html/rfc7233#section-4.1)

Comment (itamarst) [reply docs/proposed/http-storage-node-protocol.rst:515]: Yeah I was looking at MDN which is nicer, albeit less authoritative: https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests

**Review by itamarst — APPROVED** [review]
Thank you!

**Pull request merged**